### PR TITLE
Outlier detection primitive

### LINKED ISSUE #230: Isolation forest primitive
Add a primitive that supports outlier detection -  sklearn isolation forest is probably the best candidate.

### PR DESCRIPTION
**Pull request by venkatkorapaty**
fixes #230

### CODE REVIEW

File: distil/primitives/isolation_forest.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+from distil.utils import CYTHON_DEP
+import version
+
+__all__ = ("PrefeaturisedPoolingPrimitive",)
+
+logger = logging.getLogger(__name__)
+
+
+class Hyperparams(hyperparams.Hyperparams):
+    n_jobs = hyperparams.Hyperparameter[int](
```
Comment (cdbethune) [inline distil/primitives/isolation_forest.py:25]: should have lower and upper bounds defined for int/float parameters

File: test/test_isolation_forest.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            inputs=dataframe[["alpha", "bravo"]],
+        )
+        isp.fit()
+        results = isp.produce(inputs=dataframe[["alpha", "bravo"]]).value
+
+        self.assertListEqual(
+            list(results["outlier_label"]), [-1, -1, -1, -1, -1, -1, -1, -1, -1]
+        )
+
+    # def test_acled(self) -> None:
```
Comment (cdbethune) [inline test/test_isolation_forest.py:43]: Is there a reason to leave this in?

**Review by cdbethune — CHANGES_REQUESTED** [review]

File: distil/primitives/isolation_forest.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            },
+            generate_metadata=True,
+        )
+        result_df.set_index(index, inplace=True)
+
+        result_df.metadata = result_df.metadata.add_semantic_type(
+            (metadata_base.ALL_ELEMENTS, 0),
+            "https://metadata.datadrivendiscovery.org/types/PrimaryKey",
+        )
+        result_df.metadata = result_df.metadata.add_semantic_type(
```
[the remark this answers is not in the document]
Comment (venkatkorapaty) [reply distil/primitives/isolation_forest.py:135]: I'll add integer

File: distil/primitives/isolation_forest.py (hunk 1 above)
Comment (venkatkorapaty) [reply distil/primitives/isolation_forest.py:25]: for n_jobs specifically, I don't think any of the n_job hyperparams have an upper/lower bound

File: test/test_isolation_forest.py (hunk 2 above)
Comment (venkatkorapaty) [reply test/test_isolation_forest.py:43]: Was just for testing purposes, I will delete it.

**Review by cdbethune — APPROVED** [review]

**Pull request merged**